server/metric_reporter.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/plugin/segment"
+
+	"go.uber.org/zap"
+)
+
+const (
+	metricSchedulerInterval = time.Duration(24) * time.Hour
+)
+
+// MetricScheduler is the metric scheduler.
+type MetricScheduler struct {
+	l *zap.Logger
+
+	server  *Server
+	metrics api.MetricService
+}
+
+// NewMetricScheduler creates a new metric scheduler.
+func NewMetricScheduler(logger *zap.Logger, server *Server, deploymentID string) *MetricScheduler {
+	segmentService := segment.NewService(logger, server.profile.SegmentKey, deploymentID, server.store)
+
+	plan := api.FREE.String()
+	if server.subscription != nil {
+		plan = server.subscription.Plan.String()
+	}
+
+	segmentService.Identify(&api.Workspace{
```

Comment:
How about calling segmentService.Identify at the beginning of Run instead? Normally, we should avoid RPC in init code

Reply:
Updated.